.github/workflows/client-compatibility-tests.yml

```diff
@@ -38,11 +38,36 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  changes:
+    name: Find changed files
+    runs-on: ubuntu-latest
+    outputs:
+      not_ignored_changes: ${{ steps.filter.outputs.every_changed == 'true' || steps.filter.outputs.any_changed == 'true' }}
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+      - name: Filter other changes
+        uses: ./.github/actions/filter-changes
+        id: filter
+        with:
+          # Filter files that match **every** pattern in this list
+          every-filter: |
+            - '**'                     # Initially include all files
+            - '!**/*.md'               # Exclude all markdown files (lowercase)
+            - '!**/*.MD'               # Exclude all markdown files (uppercase)
```

Comment:
I'm not familiar with these tests. Is the risk here different than the core unit tests?

Reply:
@jmank88 these are end-end tests. Is there any reason we should run them on *.md changes, even if the markdown is generated? I updated this PR to only run unit tests in this case.

Reply:
I think that approach is backwards. To exclude here with confidence, we need to be able to say that no current markdown files are embedded in the image in a way that affects behavior, and none will be added in the future. That hasn't been validated and it is not enforced. It would be possible to do if we had some alignment with our `.dockerignore` file though.